### PR TITLE
bool operator for ObjectVersion

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: co/connectionSet.cpp [hunk 1]
```diff
@@ -608,29 +608,26 @@ bool ConnectionSet::_setupFDSet()
...
-        const Connection& connection = *connectionPtr.get();
-        fd.fd = connection.getNotifier();
+        fd.fd = connection->getNotifier();
 
         if( fd.fd <= 0 )
         {
-            LBINFO << "Cannot select connection " << connectionPtr
-                   << ", connection " << typeid( connection ).name()
+            LBINFO << "Cannot select connection " << connection
+                   << ", connection " << lunchbox::className( connection.get( ))
```
Comment (hernando) [inline co/connectionSet.cpp]: This may not print what you expect, before it was a reference and now it's a pointer.

File: co/objectVersion.h [hunk 2]
```diff
@@ -68,29 +68,35 @@ struct ObjectVersion
...
-                   ( identifier == rhs.identifier && version > rhs.version );
-        }
+    {
+        return identifier > rhs.identifier ||
+               ( identifier == rhs.identifier && version > rhs.version );
+    }
+
+    /** @return true if the identifier and version are set. */
+    operator bool() const
+        { return identifier != 0 && version != VERSION_NONE; }
```
Comment (hernando) [inline co/objectVersion.h]: explicit operator bool() const

https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Safe_bool

File: co/connectionSet.cpp (hunk 1 above)
Comment (eile) [inline co/connectionSet.cpp]: className() has overloads for both.

File: co/objectVersion.h (hunk 2 above)
Comment (eile) [inline co/objectVersion.h]: Done, thanks.

**Comment by hernando** [comment]
+1

**Pull request merged**